frontend/app/components/loading-button.tsx

```diff
@@ -21,52 +20,44 @@ export interface LoadingButtonProps extends ComponentProps<'button'> {
   loadingIconProps?: OmitStrict<FontAwesomeIconProps, 'icon'>;
   loadingPosition?: 'end' | 'start';
 }
-
-const LoadingButton = forwardRef<HTMLButtonElement, LoadingButtonProps>(
-  (
-    {
-      children,
-      endIcon,
-      endIconProps,
-      disabled = false,
-      loading = false,
-      loadingIcon,
-      loadingIconProps,
-      loadingPosition = 'end',
-      startIcon,
-      startIconProps,
-      ...props
-    },
+export function LoadingButton(
+  {
+    children,
+    endIcon,
+    endIconProps,
+    disabled = false,
+    loading = false,
+    loadingIcon,
+    loadingIconProps,
+    loadingPosition = 'end',
     ref,
-  ) => {
-    const resolvedLoadingIconProps = {
-      icon: loadingIcon ?? faSpinner,
-      spin: true,
-      ...(loadingIconProps ?? {}),
-    } satisfies FontAwesomeIconProps;
-
-    const isLoadingAtStartPosition = loading === true && loadingPosition === 'start';
-    const resolvedStartIcon = isLoadingAtStartPosition
-      ? resolvedLoadingIconProps
-      : { icon: startIcon, ...(startIconProps ?? {}) };
-
-    const isLoadingAtEndPosition = loading === true && loadingPosition === 'end';
-    const resolvedEndIcon = isLoadingAtEndPosition ? resolvedLoadingIconProps : { icon: endIcon, ...(endIconProps ?? {}) };
-
-    return (
-      <Button ref={ref} disabled={disabled || loading} {...props}>
-        {resolvedStartIcon.icon && (
-          <ButtonStartIcon className={loadingIcon ? '' : 'animate-spin'} {...resolvedStartIcon} icon={resolvedStartIcon.icon} />
-        )}
-        <span>{children}</span>
-        {resolvedEndIcon.icon && (
-          <ButtonEndIcon className={loadingIcon ? '' : 'animate-spin'} {...resolvedEndIcon} icon={resolvedEndIcon.icon} />
-        )}
-      </Button>
-    );
-  },
-);
-
-LoadingButton.displayName = 'LoadingButton';
-
-export { LoadingButton };
+    startIcon,
+    startIconProps,
+    ...props
+  }: LoadingButtonProps
+) {
+  const resolvedLoadingIconProps = {
+    icon: loadingIcon ?? faSpinner,
+    spin: true,
+    ...(loadingIconProps ?? {}),
+  } satisfies FontAwesomeIconProps;
+  const isLoadingAtStartPosition = loading === true && loadingPosition === 'start';
+  const resolvedStartIcon = isLoadingAtStartPosition
+    ? resolvedLoadingIconProps
+    : { icon: startIcon, ...(startIconProps ?? {}) };
+
+  const isLoadingAtEndPosition = loading === true && loadingPosition === 'end';
+  const resolvedEndIcon = isLoadingAtEndPosition ? resolvedLoadingIconProps : { icon: endIcon, ...(endIconProps ?? {}) };
+
+  return (
+    <Button ref={ref} disabled={disabled || loading} {...props}>
```

Comment:
Remove `ref` from the function parameters and pass it in as a property of `{...props}`.

Reply:
ex:

```suggestion
    <Button disabled={disabled || loading} {...props}>
```